control/iosys.py

```diff
@@ -928,18 +930,21 @@ def __init__(self, syslist, connections=[], inplist=[], outlist=[],
             sysobj_list.append(sys)
             sysname_list.append(sys.name)
 
+        # This might cause a problem if there are duplicate names...
```

Comment:
The block above makes sure there are no duplicate sysnames and you construct the statenames with a sysname prefix.

Reply:
I made a comment in the new version I just pushed.
But in short, the block above doesn't make sure there are no duplicates, it only warns the user when there are. Maybe we could extend the warning message to tell the user something like: "If you want unique names, use subsys.copy() before connecting".